### PR TITLE
small optimization in ialloc

### PR DESCRIPTION
**Pull request by theSparta**
Earlier, `bread` was being called for each inode in the range `[1, sb.ninodes]`. However, since there are `IPB` inodes in one block we should call `bread` only once per block. So, we use a block(`inum`) and offset(`i`) approach just like used in `balloc`. 

This reduces the number of calls made to `bread` by a factor `IPB`. Also, since the original code started from inode 1, we need to ignore inode 0.

### CODE REVIEW

**Comment by theSparta** [comment]
@kaashoek

**Comment by theSparta** [comment]
@kaashoek Is there a problem with this pull request?

**Comment by nubok** [comment]
I had private email correspondence with Frans Kaashoek about another potential improvement for xv6 on 19th of February. He told me that they are busy with 6.824 this semester, so they will not look into my patch in detail (though it might look promising on first sight) until they teach 6.828 again, which will be this fall. So I would rather assume that @kaashoek  simply lacks time to look into your patch.

File: fs.c [hunk 1]
```diff
@@ -185,19 +185,23 @@ static struct inode* iget(uint dev, uint inum);
...
 ialloc(uint dev, short type)
 {
-  int inum;
+  int inum, i;
   struct buf *bp;
   struct dinode *dip;
 
-  for(inum = 1; inum < sb.ninodes; inum++){
+  for(inum = 0; inum < sb.ninodes; inum += IPB){
     bp = bread(dev, IBLOCK(inum, sb));
```
Comment (dbbbit) [inline fs.c]: bread has buf cache after first real disk io, so maybe that's ok?

**Comment by kaashoek** [comment]
Thanks for the suggestion, but prefer current code for our class.

**Pull request closed without merging**